### PR TITLE
Extending CRUD operations for tptReviews

### PR DESCRIPTION
**Pull request by mbesio**

### CODE REVIEW

File: server/app.js [hunk 1]
```diff
@@ -32,16 +36,92 @@ app.get('/products/:id/reviews/', (req, res) => {
...
+  if (req.body.description !== undefined) {
+    reviewToCreate.description = req.body.description;
+  }
+  if (req.body.rating !== undefined) {
+    reviewToCreate.rating = req.body.rating;
+  }
+  if (req.body.user !== undefined) {
+    reviewToCreate.user = req.body.user;
+  }
+  model.create(reviewToCreate, (err, result) => {
```
Comment (Tina-Luk) [inline server/app.js:72]: what if the productID already exists in the database does it over ride it? or should this be an error?

File: server/app.js [hunk 2]
```diff
@@ -32,16 +36,92 @@ app.get('/products/:id/reviews/', (req, res) => {
...
+  if (req.body.title !== undefined) {
+    updateFields.title = req.body.title;
+  }
+  if (req.body.description !== undefined) {
+    updateFields.description = req.body.description;
+  }
+  if (req.body.rating !== undefined) {
+    updateFields.rating = req.body.rating;
+  }
+  model.put(reviewItem, updateFields, (err, result) => {
```
Comment (Tina-Luk) [inline server/app.js:102]: Just curious does the .put throw and error if the ID doesn't exist?

**Review by Tina-Luk — APPROVED** [review]
Liked how much separated model queries made the code more modularized. Looks good!

File: server/app.js (hunk 1 above)
Comment (mbesio) [reply server/app.js:72]: So this is the review database. Generally the productId should always exist if a user is trying t add a review. The use case is that the user is on the product page and wants to add a review (so in general the product will always exist).

The problem is that the endpoint right now does not have a way to know if a productId exists are not (it would need to verify this in the products database). Therefore you can theoretically add a review for a productId that does not exist in the product database. 

Does this address your question? 

In summary: 
if a productId exist, the endpoint would add a review for that product and display it when all reviews for that product are fetched. 
if a productId does not exist, the endpoint will still allow for a review to be posted but it will have now respective product in the db database.

File: server/app.js (hunk 2 above)
Comment (mbesio) [reply server/app.js:102]: if the reviewID does not exists, then it returns a 400 - Bad Request. Is this they way to handle this case or do you recommend a different approach?
